src/kontrol/solc_to_k.py

```diff
@@ -65,6 +65,60 @@ def solc_to_k(
     return _kprint.pretty_print(bin_runtime_definition, unalias=False) + '\n'
 
 
+@dataclass(frozen=True)
+class Input:
+    name: str
+    type: str
+    components: list[Input] = field(default_factory=list)
```

Comment:
```suggestion
    components: tuple[Input, ...] = ()
```

Reply:
Thanks! Fixed in https://github.com/runtimeverification/kontrol/pull/78/commits/dfa848b8af47b26548a846514cb4c571b638459f, together with a few other argument type- and name-related suggestions.